### PR TITLE
Support evaluation_service when using ps

### LINKED ISSUE #1468: Support evaluation_service when using ps
Currently, we set evaluation_service to None. In order to support doing evaluation when using ps, ps should send its model version to master, and master should decide whether workers should do evaluation.

### PR DESCRIPTION
**Pull request by mhaoli**
Fixes #1468.

### CODE REVIEW

File: elasticdl/proto/elasticdl.proto [hunk 1]
```diff
@@ -140,13 +140,18 @@ message ReportEvaluationMetricsRequest {
...
+}
+
 service Master {
     rpc GetTask(GetTaskRequest) returns (Task);
     rpc GetModel(GetModelRequest) returns (Model);
     rpc ReportVariable(ReportVariableRequest) returns (google.protobuf.Empty);
     rpc ReportGradient(ReportGradientRequest) returns (ReportGradientResponse);
     rpc ReportEvaluationMetrics(ReportEvaluationMetricsRequest) returns (ReportEvaluationMetricsResponse);
     rpc ReportTaskResult(ReportTaskResultRequest) returns (google.protobuf.Empty);
+    rpc report_version(ReportVersionRequest) returns (google.protobuf.Empty);
```
Comment (terrytangyuan) [inline elasticdl/proto/elasticdl.proto]: report_version -> ReportVersion to be consistent with others?

Comment (mhaoli) [reply elasticdl/proto/elasticdl.proto]: Done

File: elasticdl/python/master/evaluation_service.py [hunk 2]
```diff
@@ -220,14 +220,21 @@ def try_to_create_new_job(self):
...
         """
         Add step-based evaluation task
         """
-        model_version = self._master_servicer.get_model_version()
-        if self._eval_steps and model_version % self._eval_steps == 0:
+        if not model_version:
+            model_version = self._master_servicer.get_model_version()
+        if (
+            self._eval_steps
+            and model_version % self._eval_steps == 0
```
Comment (QiJune) [inline elasticdl/python/master/evaluation_service.py:231]: In async mode, the version in each PS may be not the same. For example, PS 0 has a version 98; PS 1 has a version 99. The evaluation step is set to 100. 

Then PS 1 reach 100, it reports to master, master will start an evaluation task. Then PS 0 reach 100 later, master will start the evaluation task again.

Comment (mhaoli) [reply elasticdl/python/master/evaluation_service.py:231]: Yes, thus I check that whether model version is bigger than the last evaluation version, please refer to 

https://github.com/sql-machine-learning/elasticdl/pull/1469/files/6238237d4165aa380497041f3646b641de0a9a7a#diff-882613b085e647dc904b962e6fd1e8c7R232

In my opinion, this condition can solve this problem.

Comment (QiJune) [reply elasticdl/python/master/evaluation_service.py:231]: Or we could check the model version in master like this:

```python
model_version = 0
i = 1
if model_version ==  i * self._eval_steps:
    i += 1
    start_evaluation()
```

Comment (QiJune) [reply elasticdl/python/master/evaluation_service.py:231]: @LiMinghao1994 Got it, thanks.

File: elasticdl/python/ps/parameter_server.py [hunk 3]
```diff
@@ -28,6 +29,10 @@ def __init__(self, args):
         self.optimizer = model_module[args.optimizer]()
         # Create Parameters instance
         self.parameters = Parameters()
+        if args.master_addr is None:
+            raise ValueError("master_addr is missing for worker")
```
Comment (QiJune) [inline elasticdl/python/ps/parameter_server.py]: master_addr is missing for parameter servers.

**Review by QiJune — APPROVED** [review]
Excellent!

File: elasticdl/python/ps/parameter_server.py (hunk 3 above)
Comment (mhaoli) [reply elasticdl/python/ps/parameter_server.py]: Done

**Pull request merged**